shared/src/commonMain/kotlin/dev/johnoreilly/confetti/ConfettiRepository.kt

```diff
@@ -1,141 +1,103 @@
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.catch
-import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.launch
-import kotlinx.datetime.LocalDate
 import kotlinx.datetime.TimeZone
 import kotlinx.datetime.toLocalDateTime
-import okio.use
 import org.koin.core.component.KoinComponent
-import org.koin.core.component.get
 import org.koin.core.component.inject
 
-class ConfettiRepository : KoinComponent {
+class ConfettiRepository(private val defaultFetchPolicy: FetchPolicy) : KoinComponent {
     val coroutineScope: CoroutineScope = MainScope()
 
-    private var apolloClient: ApolloClient? = null
     private val appSettings: AppSettings by inject()
 
-    private var refreshJob: Job? = null
+    private val apolloClientCache: ApolloClientCache by inject()
+
+    val conferenceList: Flow<List<GetConferencesQuery.Conference>> = flow {
+        val client = apolloClientCache.getClient("all")
 
-    val conferenceList: Flow<List<GetConferencesQuery.Conference>> = createApolloClient("all", false).use {
-        it.query(GetConferencesQuery()).fetchPolicy(FetchPolicy.CacheAndNetwork).toFlow()
-            .mapNotNull {
-                println(it)
+        emitAll(client.query(GetConferencesQuery()).fetchPolicy(defaultFetchPolicy)
+            .toFlow().mapNotNull {
                 it.data?.conferences
-            }.catch {
-            // do nothing
-        }
+            })
     }
 
-    private val conferenceData = MutableStateFlow<GetConferenceDataQuery.Data?>(null)
-
-    val timeZone = conferenceData.value?.config?.timezone?.let {
-        TimeZone.of(it)
-    } ?: TimeZone.currentSystemDefault()
-
-    val conferenceName = conferenceData.filterNotNull().map { it.config.name }
-
-    val sessions = conferenceData.filterNotNull().map {
-        it.sessions.nodes.map { it.sessionDetails }.sortedBy { it.startInstant }
+    // TODO: We fetch the first page only, assuming there are <100 conferences. Pagination should be implemented instead.
+    private fun conferenceQuery(conference: String): Flow<GetConferenceDataQuery.Data?> = flow {
+        emitAll(apolloClientCache.getClient(conference).query(GetConferenceDataQuery())
+            .fetchPolicy(defaultFetchPolicy)
+            .toFlow()
+            .map { it.data })
     }
 
-    val sessionsMap: Flow<Map<LocalDate, List<SessionDetails>>> = sessions.map {
-        it.groupBy {
-            it.startInstant.toLocalDateTime(
-                TimeZone.of(
-                    conferenceData.value?.config?.timezone ?: ""
-                )
-            ).date
+    fun conferenceDataFlow(): Flow<GetConferenceDataQuery.Data?> = conferenceFlow().flatMapLatest {
```

Comment:
Small comment: I'd like this to take a "conference" parameter at some point `fun conferenceDataFlow(conference: String)`. I like the idea that a given "screen" listens to only one conference. In the current state of affairs, something could call `setConference(newconference)` under the hood and change all the displayed data.

It's definitely not related to this PR, just thinking about how we want to model things. In all cases, androidx datastore looks good to me 👍 (not to be confused with the backend GCP datastore 😅 )

Reply:
I think it's the one below.  I can work harder to remove this form of it.  It should clean up the app state.

Reply:
I'm working towards this in https://github.com/joreilly/Confetti/pull/257